.test-infra/jenkins/job_LoadTests_CoGBK_Dataflow_V2_Java11.groovy

```diff
@@ -0,0 +1,260 @@
+                                "keySizeBytes": 10,
+                                "valueSizeBytes": 90,
+                                "numHotKeys": 200000
+                              }
+                            """.trim().replaceAll("\\s", ""),
+        coSourceOptions       : """
+                              {
+                                "numRecords": 2000000,
+                                "keySizeBytes": 10,
+                                "valueSizeBytes": 90,
+                                "numHotKeys": 1000
+                              }
+                            """.trim().replaceAll("\\s", ""),
+        iterations            : 4,
+        numWorkers            : 5,
+        autoscalingAlgorithm  : "NONE",
+        streaming             : isStreaming
+      ]
+
+    ],
+    [
+      title          : 'Load test: CoGBK 2GB reiteration 2MB value',
+      test           : 'org.apache.beam.sdk.loadtests.CoGroupByKeyLoadTest',
+      runner         : CommonTestProperties.Runner.DATAFLOW,
+      pipelineOptions: [
+        project               : 'apache-beam-testing',
+        region                : 'us-central1',
+        appName               : "load_tests_Java11_Dataflow_V2_${mode}_CoGBK_4",
+        tempLocation          : 'gs://temp-storage-for-perf-tests/loadtests',
+        publishToBigQuery     : true,
+        bigQueryDataset       : datasetName,
+        bigQueryTable         : "java11_dataflow_v2_${mode}_CoGBK_4",
+        influxMeasurement     : "java_${mode}_cogbk_4",
+        influxTags            : """
+                                  {
+                                    "runnerVersion": "v2",
+                                    "jdk": "java11"
+                                  }
+                                """.trim().replaceAll("\\s", ""),
+        publishToInfluxDB     : true,
+        sourceOptions         : """
+                              {
+                                "numRecords": 2000000,
+                                "keySizeBytes": 10,
+                                "valueSizeBytes": 90,
+                                "numHotKeys": 1000
+                              }
+                            """.trim().replaceAll("\\s", ""),
+        coSourceOptions       : """
+                              {
+                                "numRecords": 2000000,
+                                "keySizeBytes": 10,
+                                "valueSizeBytes": 90,
+                                "numHotKeys": 1000
+                              }
+                            """.trim().replaceAll("\\s", ""),
+        iterations            : 4,
+        numWorkers            : 5,
+        autoscalingAlgorithm  : "NONE",
+        streaming             : isStreaming
+      ]
+    ]
+  ].each { test -> test.pipelineOptions.putAll(additionalPipelineArgs) }
+}
+
+def final JOB_SPECIFIC_SWITCHES = [
+  '-Prunner.version="V2"',
+  '-PcompileAndRunTestsWithJava11',
```

Comment:
just out of curiosity - what consumes this argument? Does Gradle consume it? Does our job test framework consume it and act accordingly?

Reply:
Yes, Gradle consumes it. There are a few different places that use it, one is for picking the proper SDK container image, another is for actually compiling the code.

e.g. https://github.com/apache/beam/blob/9418994e11dbede889b508c91261ec2092f2d8a7/runners/google-cloud-dataflow-java/build.gradle#L257